src/grid_map/height_grid_map.cpp

```diff
@@ -69,7 +74,22 @@ void HeightGridMap::resize(const geometry_msgs::Vector3& min, const geometry_msg
 double HeightGridMap::rescale(nav_msgs::OccupancyGrid& map, double old_min_z, double old_max_z, double new_min_z, double new_max_z)
 {
   double old_height_scale = (old_max_z-old_min_z) / (GRID_MAP_VALUE_RANGE);
-  double new_inv_height_scale = (GRID_MAP_VALUE_RANGE) / (new_max_z-new_min_z);
+
+  if (old_height_scale < MINIMUM_SCALE)
+  {
+    old_height_scale = MINIMUM_SCALE;
+  }
+
+  double new_inv_height_scale = std::numeric_limits<double>::max();
+  if (new_max_z > new_min_z)
+  {
+    new_inv_height_scale = (std::numeric_limits<int8_t>::max()-std::numeric_limits<int8_t>::min()+1) / (new_max_z-new_min_z);
+  }
+
+  if (new_inv_height_scale > 1.0 / MINIMUM_SCALE)
+  {
+    new_inv_height_scale = 1.0 / MINIMUM_SCALE;
+  }
```

Comment:
Don't bother changing it, but a lot of these can be reduced to `value = std::min(value, min_allowed_value);`